activerecord/lib/active_record/connection_adapters/sqlite3/database_statements.rb

```diff
@@ -66,23 +56,22 @@ def execute(...) # :nodoc:
           super&.to_a
         end
 
+        def reset_isolation_level # :nodoc:
+          internal_execute("PRAGMA read_uncommitted=#{@previous_read_uncommitted}", "TRANSACTION", allow_retry: true, materialize_transactions: false)
+          @previous_read_uncommitted = nil
+        end
+
         private
           def internal_begin_transaction(mode, isolation)
             if isolation
               raise TransactionIsolationError, "SQLite3 only supports the `read_uncommitted` transaction isolation level" if isolation != :read_uncommitted
               raise StandardError, "You need to enable the shared-cache mode in SQLite mode before attempting to change the transaction isolation level" unless shared_cache?
             end
 
-            log("begin #{mode} transaction", "TRANSACTION") do
-              with_raw_connection(allow_retry: true, materialize_transactions: false) do |conn|
-                if isolation
-                  ActiveSupport::IsolatedExecutionState[:active_record_read_uncommitted] = conn.get_first_value("PRAGMA read_uncommitted")
-                  conn.read_uncommitted = true
-                end
-                result = conn.transaction(mode)
-                verified!
-                result
-              end
+            internal_execute("BEGIN #{mode} TRANSACTION", allow_retry: true, materialize_transactions: false)
```

Comment:
Shouldn’t we still pass `"TRANSACTION"` as a `name` parameter to the `#internal_execute` method?

```diff
-internal_execute("BEGIN #{mode} TRANSACTION", allow_retry: true, materialize_transactions: false)
+internal_execute("BEGIN #{mode} TRANSACTION", "TRANSACTION", allow_retry: true, materialize_transactions: false)
```

Reply:
Yes.

Reply:
https://github.com/rails/rails/pull/53308